### PR TITLE
Core/correcting warning

### PR DESCRIPTION
**Pull request by roigcarlo**
Correcting all core warnings (Clang 5). Now it compiles with -Werror.
Pending on testing it with gcc...

If you have a better suggestion of for [This one](https://github.com/KratosMultiphysics/Kratos/commit/410cf1145aed824425c6d16cba4691342853eed3) I am all ears.

### CODE REVIEW

File: kratos/sources/prime_numbers.cpp [hunk 1]
```diff
@@ -49,7 +49,7 @@ namespace Kratos
 	}
 
 
-	const std::array<std::size_t, PrimeNumbers::mNumberOfPreCalculatedPrimes> PrimeNumbers::mPrecalculatedPrimes = {
+	const std::array<std::size_t, PrimeNumbers::mNumberOfPreCalculatedPrimes> PrimeNumbers::mPrecalculatedPrimes = {{
 		1		,2		,3		,5		,7		,11		,13		,17
```
Comment (RiccardoRossi) [inline kratos/sources/prime_numbers.cpp:53]: why a double {} ?

**Comment by roigcarlo** [comment]
single { } designates an expression, which among other things can be a list, but it can be ambigous sometimes (not this particular case).

For example:
```
std::array<int> foo1 = {8, 0}
std::array<int> foo2 = {{8, 0}}
```

The first could mean size 8 with 0's, while the second will be han array of size 2 with the first elem 8 and the second 0

**Comment by RiccardoRossi** [comment]
@roigcarlo uff, but this means that one would have to define two different constructors by initializer list. Nasty...

**Comment by roigcarlo** [comment]
@RiccardoRossi Here is a more interesting description of the problem. Sorry I did not have the link yesterday: https://stackoverflow.com/questions/22501368/why-wasnt-a-double-curly-braces-syntax-preferred-for-constructors-taking-a-std

**Review by pooyan-dadvand — APPROVED** [review]
:+1:

**Pull request merged**